cmd/fs-v1.go

```diff
@@ -237,12 +237,13 @@ func (fs fsObjects) GetObject(bucket, object string, offset int64, length int64,
 		return toObjectErr(traceError(err), bucket, object)
 	}
 
-	// Reply back invalid range if the input offset and length fall out of range.
-	if offset > fi.Size || length > fi.Size {
-		return traceError(InvalidRange{offset, length, fi.Size})
+	// For negative length we read everything.
+	if length < 0 {
+		length = fi.Size
```

Comment:
Right side should be `fi.Size - offset` otherwise the next if block will wrongly error if the offset is positive.